ca/ca_test.go

```diff
@@ -491,22 +477,36 @@ func TestMultipleIssuers(t *testing.T) {
 	_, ok := ca.certProfiles.profileByName[selectedProfile]
 	test.Assert(t, ok, "Certificate profile was expected to exist")
 
-	// Test that an RSA CSR gets issuance from the RSA issuer.
+	// Test that an RSA CSR gets issuance from an RSA issuer.
 	issuedCert, err := ca.IssuePrecertificate(ctx, &capb.IssueCertificateRequest{Csr: CNandSANCSR, RegistrationID: arbitraryRegID, CertProfileName: selectedProfile})
 	test.AssertNotError(t, err, "Failed to issue certificate")
 	cert, err := x509.ParseCertificate(issuedCert.DER)
 	test.AssertNotError(t, err, "Certificate failed to parse")
-	err = cert.CheckSignatureFrom(testCtx.boulderIssuers[1].Cert.Certificate)
-	test.AssertNotError(t, err, "Certificate failed signature validation")
+	validated := false
+	for _, issuer := range ca.issuers.byAlg[x509.RSA] {
+		err = cert.CheckSignatureFrom(issuer.Cert.Certificate)
+		if err == nil {
+			validated = true
+			break
+		}
+	}
+	test.Assert(t, validated, "Certificate failed signature validation")
 	test.AssertMetricWithLabelsEquals(t, ca.signatureCount, prometheus.Labels{"purpose": "precertificate", "status": "success"}, 1)
 
```

Comment:
Can you add two additional test cases, one for RSA and the other for ECDSA, that we get an expected error if the issuers maps are ever emptied after the CA impl is created?
```
	ca.issuers.byAlg = make(map[x509.PublicKeyAlgorithm][]*issuance.Issuer)
	ca.issuers.byNameID = make(map[issuance.NameID]*issuance.Issuer)
	_, err = ca.IssuePrecertificate(ctx, &capb.IssueCertificateRequest{Csr: CNandSANCSR, RegistrationID: arbitraryRegID, CertProfileName: selectedProfile})
	test.AssertError(t, err, "Should not have issued precertificate")
```

Reply:
I don't think we particularly need to worry about the maps being emptied after they're created.

But this does raise a good question: what should the behavior be if _only_ RSA or _only_ ECDSA intermediates are configured? Should they then handle issuance for all end-entities regardless of key type?

Reply:
Okay, after internal discussion, I've added a startup-time check that there's at least one intermediate of each algorithm configured -- the CA will refuse to start if it can't issue either RSA or ECDSA end-entity certs. This isn't very flexible, but it is simple and clean and if we end up in a situation where we need to issue all end-entity certs from a single type of intermediate we can make a code change at that time.